lib/client/websocket-tracker.js

```diff
@@ -100,11 +100,16 @@ WebSocketTracker.prototype.destroy = function (cb) {
   self._onSocketDataBound = null
   self._onSocketCloseBound = null
 
-  self.socket.on('error', noop) // ignore all future errors
-  try {
-    self.socket.destroy(cb)
-  } catch (err) {
-    cb(null)
+  if (socketPool[self.announceUrl].consumers === 0) {
+    delete socketPool[self.announceUrl]
+
+    self.socket.on('error', noop) // ignore all future errors
+
+    try {
+      self.socket.destroy(cb)
+    } catch (err) {
+      if (cb) cb()
```

Comment:
`cb` is always defined, so you can get rid of the if statement. Also, `cb(null)` to be explicit.